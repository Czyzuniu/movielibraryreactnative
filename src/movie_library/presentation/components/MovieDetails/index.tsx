import {Box} from 'native-base';
import React from 'react';
import MovieBackdropImage from '../MovieBackdropImage';
import DescriptiveMovie from '../../../domain/entity/DescriptiveMovie';
import MovieDetailsHeader from '../MovieDetailsHeader';
import MovieBadgesRow from '../MovieBadgesRow';
import MovieOverview from '../MovieOverview';

type Props = {
  movie: DescriptiveMovie;
};

export default function MovieDetails({movie}: Props) {
  const {backdrop} = movie;

  return (
    <Box>
      <MovieDetailsHeader movie={movie} />
      <MovieBadgesRow movie={movie} />
      <MovieOverview movie={movie} />
      <MovieBackdropImage imagePath={backdrop} />
    </Box>
  );
}
