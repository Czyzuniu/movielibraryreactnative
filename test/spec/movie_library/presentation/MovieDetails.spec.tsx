import {render} from '@testing-library/react-native';
import React from 'react';
import ContextWrapper from '../../../utils/ContextWrapper';
import TestObjects from '../../../utils/TestObjects';
import MovieDetails from '../../../../src/movie_library/presentation/components/MovieDetails';

describe('MovieDetails Tests', () => {
  it('should render and match snapshot', () => {
    const {toJSON} = render(
      <ContextWrapper>
        <MovieDetails movie={TestObjects.DescriptiveMovie()} />
      </ContextWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
