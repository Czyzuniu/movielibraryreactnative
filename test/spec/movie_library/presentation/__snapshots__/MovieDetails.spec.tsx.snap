// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Tests should render and match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "display": "flex",
            "elevation": 2,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 4,
            "marginRight": 4,
            "marginTop": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 1.41,
          }
        }
      >
        <Image
          accessibilityLabel="movie-poster"
          alt="movie-poster"
          dataSet={Object {}}
          onError={[Function]}
          path="poster.jpg"
          resizeMode="contain"
          source={
            Object {
              "uri": "https://mock-image-api.co.uk/w500/poster.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 3,
              "height": 300,
              "maxWidth": "100%",
              "width": 300,
            }
          }
          testID="TEST_IMAGE_poster.jpg"
        />
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "marginBottom": 4,
            "marginLeft": 4,
            "marginRight": 4,
            "marginTop": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 1.41,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "700",
                "letterSpacing": 0,
                "lineHeight": 25,
                "textAlign": "center",
                "textDecorationLine": undefined,
              }
            }
          >
            this is a movie title
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Comedy
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          colorScheme="success"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#404040",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
              "paddingBottom": 10,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 10,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#22c55e",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                AddToFavourites
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "flex": 1,
          "height": 90,
          "marginBottom": 6,
          "marginLeft": 6,
          "marginRight": 6,
          "marginTop": 6,
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#fafafa",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "textDecorationLine": undefined,
                }
              }
            >
              8
            </Text>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#d97706",
                    "fontSize": 12,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#fafafa",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "textDecorationLine": undefined,
                }
              }
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#d97706",
                    "fontSize": 12,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Runtime
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#fafafa",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "textDecorationLine": undefined,
                }
              }
            >
              Apr, 24 2022
            </Text>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#d97706",
                    "fontSize": 12,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "flex": 1,
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#fafafa",
            "fontFamily": undefined,
            "fontSize": null,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 30,
            "marginBottom": 4,
            "marginLeft": 4,
            "marginRight": 4,
            "marginTop": 4,
            "textDecorationLine": undefined,
          }
        }
      >
        Overview
      </Text>
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#fafafa",
            "fontFamily": undefined,
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
            "marginBottom": 4,
            "marginLeft": 4,
            "marginRight": 4,
            "marginTop": 4,
            "textAlign": "left",
            "textDecorationLine": undefined,
          }
        }
      >
        this is an overview
      </Text>
    </View>
    <Image
      accessibilityLabel="movie-backdrop-image"
      alt="movie-backdrop-image"
      dataSet={Object {}}
      onError={[Function]}
      path="backdrop.jpg"
      resizeMode="cover"
      source={
        Object {
          "uri": "https://mock-image-api.co.uk/w500/backdrop.jpg",
        }
      }
      style={
        Object {
          "height": 224,
          "maxWidth": "100%",
          "width": "100%",
        }
      }
      testID="TEST_IMAGE_backdrop.jpg"
    />
  </View>
</RNCSafeAreaProvider>
`;
